.changesets/fix_geal_fix_compression_finish.md

```diff
@@ -0,0 +1,5 @@
+### make sure the compression state is flushed ([Issue #3035](https://github.com/apollographql/router/issues/3035))
+
+In some cases, the "finish" call to flush the compression state at the end does not flush th entire state, so it has to be called multiple times.
+
+By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/3037
```

Comment:
I would suggest calling out that this was a regression introduced in 1.16.0 and linking to the issue/PR that introduced it — for posterity — and so folks immediately recognize that 1.16.0 has its limitations.